internal/api/installation.go

```diff
@@ -128,16 +129,16 @@ func handleCreateInstallation(c *Context, w http.ResponseWriter, r *http.Request
 	}
 
 	installation := model.Installation{
-		OwnerID:       createInstallationRequest.OwnerID,
+		OwnerID:       strings.TrimSpace(createInstallationRequest.OwnerID),
 		GroupID:       &createInstallationRequest.GroupID,
-		Version:       createInstallationRequest.Version,
-		Image:         createInstallationRequest.Image,
-		DNS:           createInstallationRequest.DNS,
-		Database:      createInstallationRequest.Database,
-		Filestore:     createInstallationRequest.Filestore,
-		License:       createInstallationRequest.License,
-		Size:          createInstallationRequest.Size,
-		Affinity:      createInstallationRequest.Affinity,
+		Version:       strings.TrimSpace(createInstallationRequest.Version),
+		Image:         strings.TrimSpace(createInstallationRequest.Image),
+		DNS:           strings.TrimSpace(createInstallationRequest.DNS),
+		Database:      strings.TrimSpace(createInstallationRequest.Database),
+		Filestore:     strings.TrimSpace(createInstallationRequest.Filestore),
+		License:       strings.TrimSpace(createInstallationRequest.License),
+		Size:          strings.TrimSpace(createInstallationRequest.Size),
+		Affinity:      strings.TrimSpace(createInstallationRequest.Affinity),
```

Comment:
Hmm, instead of trimming spaces we should handle this in validation: https://github.com/mattermost/mattermost-cloud/blob/master/model/installation_request.go#L56-L61

I think we shouldn't silently succeed in creating a new installation when the dns entry had some bad characters. We should error and let the sender fix it.

Reply:
removed